### PR TITLE
Add test cases for newline handling in form payloads

### PR DESCRIPTION
**Pull request by andreubotella**
(The title of this PR used to be "Add test cases for urlencoded serialization of filenames", dealing in particular with newline normalizations in the urlencoded serialization. However, the scope of the spec issues whatwg/url#562 and whatwg/html#6247 grew in scope as similar cases were found in other form enctypes.)

### CODE REVIEW

**Review by annevk — APPROVED** [review]
You could make this a window.js test, but fine either way.

**Review by annevk — CHANGES_REQUESTED** [review]
I initially thought it was okay, but this is also testing form submission and that currently requires newline normalization for file names that the expected results do not seem to reflect.

And since this is really about form submission perhaps this should be placed in the `html/` top-level directory, though I'm not entirely sure as there is no other way to test the encoding aspect of urlencoding... Either way we should probably link from a README on the other side, wherever we decide to put these.

**Comment by annevk** [comment]
Although this matches implementations per https://github.com/whatwg/url/issues/562#issuecomment-738276174, I want to wait with landing these tests until the specifications back them up. Or am I misunderstanding something still?

**Comment by andreubotella** [comment]
I wrote these tests are written to follow the specs, without the newline normalization, so that all browsers fail. Since we now have settled on a way to fix the spec to match the implementations, I'll update the tests to match.

**Comment by annevk** [comment]
Oh right, so much misunderstanding around these newlines. 😊

Are you also planning to test `text/plain`? Do you want help with the specification changes?

**Comment by andreubotella** [comment]
Should I add the tests for `text/plain` to this PR?

**Comment by annevk** [comment]
Sounds great!

**Review by annevk — APPROVED** [review]
This looks great! I can work on spec changes next week, but also happy to defer to you. Let me know.

**Comment by andreubotella** [comment]
I'll open the PRs on HTML and XHR (for `FormData`), if you don't mind.

**Comment by annevk** [comment]
That sounds great, thanks! We'll need one for URL too as it'll no longer get files.

**Comment by andreubotella** [comment]
Since the last post in this PR, we realized that the proposed changes don't work (see https://github.com/whatwg/url/issues/562#issuecomment-744453348), and the changes grew in scope to be about newline normalization in all form submission enctypes. Although the exact behaviors are still being decided in whatwg/html#6247, this PR tests for things that are very unlikely to change. But it wouldn't be enough to test

Should this PR be repurposed to fit whatwg/html#6247, or should I open a new one for the remaining tests?

**Comment by annevk** [comment]
Hey @andreubotella, what do you mean with "but it wouldn't be enough to test"? I think it's up to you to decide how to proceed. I'm happy to review the tests all in one go and I think @domenic would be okay with that as well.

**Comment by andreubotella** [comment]
I think this is now ready for review. For any reviewers: note that the spec PR corresponding to these tests is whatwg/html#6287, and see also whatwg/html#6247 for some context.

**Comment by andreubotella** [comment]
Since it seems like implementations agree on the spec change, we should work towards getting this merged. Since this PR touched many files all over the place, there are many suggested reviewers, but in addition to @annevk, I'd like to get @tkent-google's approval, since in #26838 I changed the tests for form-associated custom elements to match the spec, not yet realizing that both Chrome and Firefox [normalized strings added in the `formdata` event in the same way](https://github.com/whatwg/html/issues/6247) (and Safari only didn't because it doesn't implement that event). This PR changes that back and also adds tests for files in FACEs.

**Review by annevk — APPROVED** [review]
I'm happy with these tests. Thanks so much for taking the time to write them all! One thing that would have made reviewing slightly easier is if the function was defined before all the invocations to it, but it doesn't matter much.

**Comment by andreubotella** [comment]
@tkent-google Ping

File: html/semantics/forms/form-submission-0/multipart-formdata.window.js
Comment (tkent-google) [inline html/semantics/forms/form-submission-0/multipart-formdata.window.js:317]: This function looks to be sharable with text-plain.window.js and urlencoded2.window.js. Please make a helper js file for this function, and refer to it in these tests by "// META: script=".

Comment (andreubotella) [reply html/semantics/forms/form-submission-0/multipart-formdata.window.js]: Done. Other than just the code duplication in the `form` function being repeated with minor changes across the three test files, the contents of the `form` function itself were duplicated for each of the runs of each test (what I'm calling "normal form" vs "`formdata` event"), so I also moved that code to a separate function.

**Review by tkent-google — APPROVED** [review]
LGTM

**Pull request merged**
